src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -11,22 +11,51 @@
 namespace Microsoft.CodeAnalysis.CSharp
 {
     /// <summary>
-    /// This portion of the binder converts deconstruction-assignment syntax (AssignmentExpressionSyntax nodes with the left being a tuple expression)
-    /// into a BoundDeconstructionAssignmentOperator (or bad node).
+    /// This portion of the binder converts deconstruction-assignment syntax (AssignmentExpressionSyntax nodes with the left
+    /// being a tuple expression or declaration expression) into a BoundDeconstructionAssignmentOperator (or bad node).
     /// </summary>
     internal partial class Binder
     {
-        private BoundExpression BindDeconstructionAssignment(AssignmentExpressionSyntax node, DiagnosticBag diagnostics)
+        private BoundExpression BindDeconstruction(AssignmentExpressionSyntax node, DiagnosticBag diagnostics)
         {
-            var left = (TupleExpressionSyntax)node.Left;
-            ArrayBuilder<DeconstructionVariable> checkedVariables = BindDeconstructionAssignmentVariables(left.Arguments, left, diagnostics);
+            var left = node.Left;
+            var right = node.Right;
 
+            if (node.IsDeconstructionDeclaration())
+            {
+                return BindDeconstructionDeclaration(node, left, right, diagnostics);
+            }
+
+            // We only parse assignment-only or declaration-only deconstructions at this point
+            AssertDeconstructionIsAssignment(left);
```

Comment:
Is the comment correct? Given the assert, it looks like we're only expecting to bind (rather than parse) assignments, not declarations.